v2/googlecloud-to-googlecloud/src/main/java/com/google/cloud/teleport/v2/templates/bigtablechangestreamstopubsub/FailsafeModJsonToPubsubMessageTransformer.java

```diff
@@ -0,0 +1,311 @@
+            publisher.awaitTermination(5, TimeUnit.MINUTES);
+          }
+        } catch (Exception e) {
+          throw new RuntimeException(e);
+        }
+      }
+
+      public static ChangelogEntry mapJsonToChangelogEntry(String jsonString) throws Exception {
+        Schema schema = ChangelogEntry.getClassSchema();
+
+        // Create a Decoder for JSON data
+        Decoder decoder = DecoderFactory.get().jsonDecoder(schema, jsonString);
+
+        DatumReader<ChangelogEntry> reader = new SpecificDatumReader<>(schema);
+
+        ChangelogEntry changelogEntry = reader.read(null, decoder);
+        return changelogEntry;
+      }
+
+      public static ChangelogEntryProto mapJsonToChangelogEntryProto(String jsonString)
+          throws Exception {
+        ChangelogEntryProto.Builder builder = ChangelogEntryProto.newBuilder();
+        JsonFormat.parser().ignoringUnknownFields().merge(jsonString, builder);
+
+        // Build the protobuf message
+        ChangelogEntryProto changelogEntryProto = builder.build();
+        return changelogEntryProto;
+      }
+
+      @ProcessElement
+      public void processElement(ProcessContext context) {
+        LOG.info("Reading a failsafeModJsonString");
+        FailsafeElement<String, String> failsafeModJsonString = context.element();
+
+        LOG.info("");
+
+        try {
+          com.google.pubsub.v1.PubsubMessage pubSubMessage =
+              publishModJsonStringToPubsubMessage(failsafeModJsonString.getPayload());
+
+          context.output(pubSubMessage);
+        } catch (Exception e) {
+          LOG.error("Writing to transform dead letter out", e);
+          context.output(
+              transformDeadLetterOut,
+              FailsafeElement.of(failsafeModJsonString)
+                  .setErrorMessage(e.getMessage())
+                  .setStacktrace(Throwables.getStackTraceAsString(e)));
+        }
+      }
+
+      /* Schema Details:  */
+      private PubsubMessage publishModJsonStringToPubsubMessage(String modJsonString)
+          throws Exception {
+        String messageFormat = pubSubUtils.getDestination().getTopicMessageFormat();
+        String messageEncoding = pubSubUtils.getDestination().getTopicMessageEncoding();
+        Publisher publisher = null;
+        ByteArrayOutputStream byteStream = new ByteArrayOutputStream();
+
+
+        block:
+
+        switch (messageFormat) {
+          case "AVRO":
+            ChangelogEntry changelogEntry = mapJsonToChangelogEntry(modJsonString);
+            Encoder encoder = null;
+            switch (messageEncoding) {
+              case "BINARY":
+                encoder = EncoderFactory.get().directBinaryEncoder(byteStream, /* reuse= */ null);
+                break;
+              case "JSON":
+                encoder =
+                    EncoderFactory.get().jsonEncoder(ChangelogEntry.getClassSchema(), byteStream);
+                break;
+              default:
+                break block;
+            }
```

Comment:
looks like it can be made a separate method which would throw exception is message encoding is not supported (something we should validate ahead of time)